inference-engine/thirdparty/clDNN/tests/test_cases/reorder_gpu_test.cpp

```diff
@@ -23,8 +23,251 @@ using namespace cldnn;
 using namespace tests;
 using namespace testing;
 
-TEST(reorder_gpu_f32, basic)
+static void compare_bfyx2blocked_with_ref(const data_types input_data_type, const data_types output_data_type,
+    cldnn::format input_format, cldnn::format output_format,
+    int32_t b_in, int32_t f_in, int32_t x_in, int32_t y_in, int32_t z_in = 0, int32_t w_in = 0)
 {
+    const auto& engine = get_test_engine();
+
+    tensor ts;
+    if (input_format == format::bfyx) {
+        ts = { b_in, f_in, x_in, y_in };
+    }
+    else if (input_format == format::bfzyx) {
+        ts = { b_in, f_in, x_in, y_in, z_in };
+    }
+    else {
+        ts = { b_in, f_in, x_in, y_in, z_in, w_in };
+    }
+
+    auto input = memory::allocate(engine, { input_data_type, input_format, ts });
+    layout output_layout(output_data_type, output_format, ts);
+
+    auto input_ptr = input.pointer<unsigned char>();
+    unsigned char i = 1;
+    for (auto it = input_ptr.begin(); it != input_ptr.end(); ++it)
+    {
+        *it = (i++);
+        if (i > 100) {
+            i = 1;
+        }
+    }
+
+    // run on reference(reorder_data) kernel
+    topology topology_ref(
+        input_layout("input", input.get_layout()),
+        reorder("reorder", "input", output_layout));
+
+    cldnn::build_options options_ref;
+    cldnn::implementation_desc reorder_ref = { output_format, "reorder_data" };
+    options_ref.set_option(cldnn::build_option::force_implementations({ {"reorder", reorder_ref} }));
+
+    network network_ref(engine, topology_ref, options_ref);
+    network_ref.set_input_data("input", input);
+
+    std::map<cldnn::primitive_id, cldnn::network_output> outputs_ref;
+
+    outputs_ref = network_ref.execute();
+    cldnn::event e1 = outputs_ref.at("reorder").get_event();
+    e1.wait();
+
+    auto output_ref = outputs_ref.begin()->second.get_memory();
+    auto output_ref_ptr = output_ref.pointer<unsigned char>();
+
+    // run on optimized(reorder_data_bfyx_to_blocked_format) kernel
+    topology topology(
```

Comment:
you can share the topology to the two networks

Reply:
Applied.